wgsl/index.bs

```diff
@@ -4320,6 +4320,39 @@ See [[#builtin-inputs-outputs]] for how to declare a built-in variable.
       <td width="50%">The current invocation's [=global invocation ID=],
             i.e. its position in the [=compute shader grid=].
 
+  <tr><td>`sample_index`
+      <td>fragment
+      <td>in
+      <td>u32
+      <td width="50%">Sample index for the current fragment.
+         The value is least 0 and at most `sampleCount`-1, where
+         [[WebGPU#dom-gpurenderpipelinedescriptor-samplecount|sampleCount]]
+         is the number of MSAA samples specified for the GPU render pipeline.
+         <br>See [[WebGPU#gpurenderpipe|WebGPU &sect; GPURenderPipeline]].
+
+  <tr><td>`sample_mask_in`
+      <td>fragment
+      <td>in
+      <td>u32
+      <td width="50%">Sample coverage mask for the current fragment.
+         It contains a bitmask indicating which samples in this fragment are covered
+         by the primitive being rendered.
+         <br>See [[WebGPU#sample-masking|WebGPU &sect; Sample Masking]].
+
+  <tr><td>`sample_mask`
+      <td>fragment
+      <td>out
+      <td>u32
+      <td width="50%">Sample coverage mask control for the current fragment.
+         The last value written to this variable becomes the
+         [[WebGPU#shader-output-mask|shader-output mask]].
+         Zero bits in the written value will cause corresponding samples in
+         the color attachments to be discarded.
+         <br>The value in the variable is significant only if the
+         `sample_mask` variable is [=statically accessed=]
+         by the fragment shader. If the variable is not statically accessed,
+         then other factors determine sample coverage.
+         <br>See [[WebGPU#sample-masking|WebGPU &sect; Sample Masking]].
```

Comment:
There's an interesting case to think about:  What is the initial value of sample_mask?

Our current rules have it default-initialized to 0. That is consistent but a foot-gun.
I thought about make a special rule to say it's initialized to all-1-bits, but then I didn't want to force backends to do an initial store, as that might mess up semantics somewhere along the way.

Reply:
We should in general approach the topic of zero-initialization differently for built-in outputs. Sample mask is one example, but even just the output position is somewhat useless to initialize to all 0s.